### PR TITLE
Make freqs_cis a persistent buffer for pp init

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #161
* #172
* #203
* #202
* __->__ #201

currently, planning to use a 'seed checkpoint' to initialize the
pipeline parallel model chunks after moving them from meta device to
cuda/empty.

non-persistent buffers are incompatible with this approach, as they are
missing from the checkpoint and thus require manual init.

an alternative is to manually run the initializer for just the
non-persistent buffers after loading a seed-checkpoint, but this
approach is nearly equivalent with less code changes.

### CODE REVIEW

File: torchtrain/models/llama/model.py [hunk 1]
```diff
@@ -313,9 +313,7 @@ def __init__(self, model_args: ModelArgs):
         super().__init__()
         self.model_args = model_args
         self.tok_embeddings = nn.Embedding(model_args.vocab_size, model_args.dim)
-        self.register_buffer(
-            "freqs_cis", self._precompute_freqs_cis(), persistent=False
-        )
+        self.register_buffer("freqs_cis", self._precompute_freqs_cis(), persistent=True)
```
Comment (awgu) [inline torchtrain/models/llama/model.py:324]: Should we include a comment with https://github.com/pytorch/pytorch/issues/123411 to mention that we are making this `persistent=True` since that is required for PP?
